### ISSUE TITLE
When setting use_vllm=True, there is an explosive increase in memory usage.

### ISSUE THREAD

**Issue by Tuziking** [huggingface/trl#3124]
I tried training the Qwen2.5-7B model with use_vllm=True. I found that after the first step, the GPU memory usage was only 50% (H20). However, when the second step started, the memory usage suddenly spiked to full capacity, leading to an OOM error. In contrast, when I set use_vllm=False, the training proceeded smoothly and the memory usage increased gradually. My config is as follow:
```
model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=torch.bfloat16, _attn_implementation="sdpa")
    lora_config = LoraConfig(
        r=32,  
        lora_alpha=256,  
        target_modules=["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "up_proj", "down_proj"],
        lora_dropout=0.1, 
        task_type=TaskType.CAUSAL_LM
    )
    model = get_peft_model(model, lora_config)


    training_args = GRPOConfig(
        output_dir=output_dir,
        run_name=run_name,
        learning_rate=5e-6,
        adam_beta1 = 0.9,
        adam_beta2 = 0.99,
        weight_decay = 0.1,
        warmup_ratio = 0.1,
        lr_scheduler_type='cosine',
        logging_steps=1,
        bf16=True,
        per_device_train_batch_size=2,
        gradient_accumulation_steps=1,
        num_generations=4,
        max_prompt_length=512,
        max_completion_length=2048,
        num_train_epochs=1,
        save_steps=100,
        max_grad_norm=0.1,
        log_on_each_node=False,
        # use_vllm=True,
        # vllm_device="cuda:0",
        report_to="wandb",
        # vllm_gpu_memory_utilization=0.5,
    )
```
System Info：
```
trl=0.15.2
vllm=0.7.2
```

**Comment by albertvillanova** [comment]
Hey @Tuziking, sorry for the very late reply — we're catching up on the backlog. 

There isn't quite enough here to investigate: the shared config has `use_vllm`, `vllm_device` and `vllm_gpu_memory_utilization` all commented out and no dataset/reward/trainer, so it isn't runnable; the System Info is just two version numbers rather than full `trl env` output; and there's no traceback for the OOM. On top of that, the GRPO + vLLM integration has been substantially reworked since 0.15.2 (we're now on v1.8.0): `vllm_device` was removed and replaced by `vllm_mode` (`"colocate"` default vs `"server"`), with memory controls like `vllm_gpu_memory_utilization` and `vllm_enable_sleep_mode` (see https://github.com/huggingface/trl/blob/3af007ecd3a791b7d16a16f3d11934bf88e22398/trl/trainer/grpo_config.py#L121-L129). 

Upgrading to a current TRL will very likely change or resolve the second-step memory spike you saw. 

Closing for now, but happy to re-open if it still happens on the latest version — please share a complete runnable script, full `trl env` output, and the traceback.
